core/templates/pages/exploration-player-page/services/question-player-engine.service.ts

```diff
@@ -96,34 +87,35 @@ export class QuestionPlayerEngineService {
       successCallback: (initialCard: StateCard, nextFocusLabel: string) => void,
       errorCallback: () => void): void {
     if (!this.questions || this.questions.length === 0) {
+      this.alertsService.addWarning('There are no questions to display.');
       errorCallback();
       return;
     }
     this.contextService.setCustomEntityContext(
       AppConstants.ENTITY_TYPE.QUESTION, this.questions[0].getId());
-    var initialState = this.questions[0].getStateData();
+    let initialState = this.questions[0].getStateData();
 
-    var questionHtml = this.makeQuestion(initialState, []);
+    let questionHtml = this.makeQuestion(initialState, []);
     if (questionHtml === null) {
-      this.alertsService.addWarning('Expression parsing error.');
+      this.alertsService.addWarning('Question name should not be empty.');
       return;
     }
 
-    this.currentIndex = 0;
+    this.setCurrentIndex(0);
     this.nextIndex = 0;
 
-    var interaction = initialState.interaction;
-    var nextFocusLabel = this.focusManagerService.generateFocusLabel();
+    let interaction = initialState.interaction;
+    let nextFocusLabel = this.focusManagerService.generateFocusLabel();
 
-    var interactionId = interaction.id;
-    var interactionHtml = null;
+    let interactionId = interaction.id;
+    let interactionHtml = null;
```

Comment:
Can we use `const` for these? It seems like these variables are not reassigned values after being initialized.

Reply:
Ditto elsewhere in this file.

Reply:
done.